features/update/worker_restart/feature.java

```diff
@@ -0,0 +1,147 @@
+package update.worker_restart;
+
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import io.temporal.sdkfeatures.Feature;
+import io.temporal.sdkfeatures.Run;
+import io.temporal.sdkfeatures.Runner;
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.UpdateMethod;
+import io.temporal.workflow.Workflow;
+import org.junit.jupiter.api.Assertions;
+import update.updateutil.UpdateUtil;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+@ActivityInterface
+public interface feature extends Feature, IntWorkflow {
+
+  @ActivityMethod
+  void block();
+
+  @UpdateMethod()
+  int update(int i);
+
+  @SignalMethod
+  void finish();
+
+  class Impl implements feature {
+
+    private boolean doFinish = false;
+    private int counter = 0;
+
+    static Object updateStartedLock = new Object();
+    static Boolean updateStarted = false;
+
+    static Object updateContinueLock = new Object();
+    static Boolean updateContinue = false;
+
+    private static void signalUpdateStarted() {
+      synchronized (updateStartedLock) {
```

Comment:
This isn't the best wake up approach in modern Java IMO. Consider CompletableFuture, Condition, Semaphore, etc.

Reply:
What your saying my university Java course failed me 😮, I replaced with semaphores.